### ISSUE TITLE
[Feature Request] Auto/manual scan mode option

### ISSUE THREAD

**Issue by eleius** [zyrouge/symphony#218]
### Description

Every time I start the app it has to reindex my music and the "Loading" message lasts for several minutes. It also happens if I simply close and reopen the app without changing anything.

### Solution

It would be nice to have a setting to change scan mode to automatic or manual, so that when set to "manual" I can simply browse the folder I want.

I assume metadata is required to browse songs by title, author and genre, in that case setting indexing to "manual" could simply limit the available options to Folders and Playlists while hiding the others.

### Checklist

-   [x] I have read the [Contribution Guidelines](https://github.com/zyrouge/symphony/wiki/Contributions-Guidelines#issues).

**Comment by zyrouge** [comment]
Re-indexing should take that long since the metadata is cached by both Symphony and Media Store. This seems to be a weird behaviour. Once the complete metadata is done scanning, the next scans shouldn't take the same amount of time, it should be pretty much instant. Would like more information on this.

**Comment by eleius** [comment]
What information do you need? The app is reading files from an external 512GB samsung evo sd card where I keep several folders (books, videos, etc.) while all songs are in a directory with several subdirectories. Maybe it tries to scan the whole sd card looking for files and needs time?

**Comment by eleius** [comment]
I forgot to mention I'm already using the "Folder" view so I assumed browsing files wouldn't need scanning, hence my request.

**Comment by zyrouge** [comment]
Try letting it scan completely, close and reopen the app. Does it take the same amount of time to scan again?

**Comment by eleius** [comment]
Yes. I tried again now, and also tried again after restarting the device.
It starts with "damn this is so empty" and then I can see the music folder being "populated" one subdirectory at a time while the loading icon spins.

**Comment by zyrouge** [comment]
Implementing this solution is very tedious. Categorizing does not happen unless the songs are known, so unless it indexes the song, it is not possible to categorize it in it's own sub-directory. So, I think it's better to investigate why this rather occurs or to avoid indexing (or re-indexing) again.

**Comment by eleius** [comment]
> Categorizing does not happen unless the songs are known, so unless it indexes the song, it is not possible to categorize it in it's own sub-directory.

In fact I was asking for a simple "file browser" mode, without categorizing, metadata, etc. I should just show available files like the OS file manager does and, once a song has been selected, it can then load and display its metadata. Just an idea though, don't know if it can be done easily.

**Comment by zyrouge** [comment]
Implementation of that requires remaking most the internal services since it requires lazy loading metadata.

**Comment by eleius** [comment]
How else can i help?

**Comment by zyrouge** [comment]
Probably check if it's an indexing issue would help. I'm also not sure if it's cause of reading from an SD card.
